### PR TITLE
[WIP] Move to pull-streams

### PR DESCRIPTION
**Pull request by dignifiedquire**
- [x] Migrate the repo under `test/go-ipfs-repo` to the latest version of go-ipfs. 

### Tests Passing

- [x] Core
  - [x] both
    - [x] bitswap
    - [x] block
    - [x] bootstrap
    - [x] config
    - [x] files
    - [x] generic
    - [x] object
    - [x] init
  - [x] Node
    - [x] init
    - [x] swarm
  - [x] Browser
    - [x] bitswap
    - [x] everything else
- [x] http
  - [x] bitswap
  - [x] block
  - [x] bootstrap
  - [x] config
  - [x] files
  - [x] id
  - [x] object
  - [x] repo
  - [x] swarm
  - [x] version
- [x] cli
  - [x] bitswap
  - [x] block
  - [x] bootstrap
  - [x] commands
  - [x] config
  - [x] bootstrap
  - [x] files
  - [x] id
  - [x] object
  - [x] swarm

Ref #403

### CODE REVIEW

**Comment by dignifiedquire** [comment]
It looks like the swarm tests are not passing due to secio. Next step, test secio better and fix any issues with it + interop with go.

**Comment by dignifiedquire** [comment]
I finally have the swarm tests passing. It seems there is something strange going on on the connection between nodes which results in the bitwap tests not passing. Other than those, only some smaller details in the `files` api need to be ironed out :)

**Comment by dignifiedquire** [comment]
bitswap tests pass with all new bug fixes in place :)

**Comment by diasdavid** [comment]
@dignifiedquire so, it is only missing the files tests? :D

File: package.json [hunk 1]
```diff
@@ -93,12 +93,18 @@
     "peer-id": "^0.7.0",
     "peer-info": "^0.7.0",
     "promisify-es6": "^1.0.1",
+    "pull-file": "^0.5.0",
+    "pull-paramap": "^1.1.6",
+    "pull-stream": "^3.4.3",
+    "pull-stream-to-stream": "^1.3.1",
```
Comment (diasdavid) [inline package.json]: Let's make sure to update it to latest

File: src/core/ipfs/block.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ module.exports = function block (self) {
   return {
     get: (hash, callback) => {
       hash = cleanHash(hash)
-
-      self._blockS.getBlock(hash, callback)
+      console.log('block.get', hash)
```
Comment (diasdavid) [inline src/core/ipfs/block.js]: - [ ] clean

**Comment by dignifiedquire** [comment]
pretty much only a single file test is failing at this point. Which is only due to the to node stream conversions

**Comment by diasdavid** [comment]
@dignifiedquire could you describe it more?

**Comment by dignifiedquire** [comment]
The get directory test fails due to the content stream of one nested file not ending properly for some reason. I haven't tracked down where exactly it is failing.

**Comment by dignifiedquire** [comment]
With the latest version I finally have them file tests passing when using https://github.com/pull-stream/pull-stream-to-stream/pull/5

**Comment by dignifiedquire** [comment]
All tests passing locally with https://github.com/ipfs/interface-ipfs-core/pull/70 🎉

**Pull request merged**